packages/pages/src/common/src/parsers/templateParser.ts

```diff
@@ -0,0 +1,32 @@
+import SourceFileParser, { createTsMorphProject } from "./sourceFileParser.js";
+
+/**
+ * TemplateParser is a class for parsing the template.
```

Comment:
Add tests.

Reply:
Similar to my other comment, it'd be ideal if makeClientTemplate didn't save the change automatically, but instead returned the results. This way we can add tests that use actual templates that include a bunch of other cruft a user could add so that we can verify that the client output is what we'd expect. Right now all of the tests for the other file are very limited with their input. Adding a test for this would give us more confidence that it does exactly what we want.

Reply:
So are you going to do this in the next PR?

Reply:
Oh hmmm I thought I did it here? By creating templateParser.test.ts and changing how templateParser's makeClientTemplate returns

Reply:
Gotcha. Can you add some tests that use full template files too? That will help us feel more confident everything works when there's more than just one line input.